beam/core/src/main/java/cz/o2/proxima/beam/core/io/AbstractRepositoryCoder.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.beam.core.io;
+
+import cz.o2.proxima.repository.Repository;
+import cz.o2.proxima.repository.RepositoryFactory;
+import org.apache.beam.sdk.coders.CustomCoder;
+
+abstract class AbstractRepositoryCoder<T> extends CustomCoder<T> {
```

Comment:
should coders be thread safe or not?

Reply:
I think that single instance of coder should be accessed from single thread only. But it is possible that there exist multiple instance of the same coder at the same time, so static coders can be used only when they are thread safe.